go/vt/vttablet/onlineddl/vrepl.go

```diff
@@ -289,6 +289,9 @@ func (v *VRepl) generateFilterQuery() error {
 			sb.WriteString(fmt.Sprintf("CONCAT(%s)", escapeName(name)))
 		case sourceCol.Type() == "json":
 			sb.WriteString(fmt.Sprintf("convert(%s using utf8mb4)", escapeName(name)))
```

Comment:
@dbussink do you think this is still needed? I don't *think* so anymore, now that we have native JSON type support.

Reply:
```
❯ mysql commerce -e "create table json_test (id int not null primary key, j1 json); insert into json_test values (1, '{\"name\":\"Matt\"}')"
❯ mysql commerce -e "insert into json_test select id+10, j1 from json_test"
❯ mysql commerce -e "select * from json_test" --column-type-info
Field   1:  `id`
Catalog:    `def`
Database:   `commerce`
Table:      `json_test`
Org_table:  `json_test`
Type:       LONG
Collation:  binary (63)
Length:     11
Max_length: 2
Decimals:   0
Flags:      NOT_NULL PRI_KEY NO_DEFAULT_VALUE NUM PART_KEY

Field   2:  `j1`
Catalog:    `def`
Database:   `commerce`
Table:      `json_test`
Org_table:  `json_test`
Type:       JSON
Collation:  binary (63)
Length:     4294967295
Max_length: 16
Decimals:   0
Flags:      BLOB BINARY


+----+------------------+
| id | j1               |
+----+------------------+
|  1 | {"name": "Matt"} |
| 11 | {"name": "Matt"} |
```

I expect this to be bytes we pass on to MySQL "on the other side" and they are interpreted there as either a JSON field or serialized as a utf8mb4 string if some other type on the target.

Reply:
I don't think it's a major deal on the source side as the primary problems we've seen there are when these `CONVERT` calls then preclude us from using the desired index in the `rowstreamer` query and you can't add indexes directly on JSON columns anyway.

Reply:
Let's leave it like so for now.

Reply:
JSON is a bit special anyway, since we can't use the direct textual representation, but we turn it into a sql expression using `JSON_OBJECT` so we lose as little type information as possible.